demisto_sdk/yaml_tools/update_generic_yml.py

```diff
@@ -0,0 +1,145 @@
+
+    def get_id_and_version_path_object(self):
+        """Gets the dict that holds the id and version fields.
+
+        Returns:
+            Dict. Holds the id and version fields.
+        """
+        yml_type = self.__class__.__name__
+        path = self.ID_AND_VERSION_PATH_BY_YML_TYPE[yml_type]
+        return self.yml_data.get(path, self.yml_data)
+
+    def remove_copy_and_dev_suffixes_from_name(self):
+        """Removes any _dev and _copy suffixes in the file.
+
+        When developer clones playbook/integration/script it will automatically add _copy or _dev suffix.
+        """
+        print(F'Removing _dev and _copy suffixes from name and display tags')
+
+        self.yml_data['name'] = self.yml_data.get('name', '').replace('_copy', '').replace('_dev', '')
+        if self.yml_data.get('display'):
+            self.yml_data['display'] = self.yml_data.get('display', '').replace('_copy', '').replace('_dev', '')
+
+    def update_id_to_equal_name(self):
+        """Updates the id of the YML to be the same as it's name."""
+        print(F'Updating YML ID to be the same as YML name')
+
+        self.id_and_version_location['id'] = self.yml_data['name']
+
+    def set_version_to_default(self):
+        """Replaces the version of the YML to default."""
+        print(F'Setting YML version to default: {self.DEFAULT_YML_VERSION}')
+
+        self.id_and_version_location['version'] = self.DEFAULT_YML_VERSION
+
+    def save_yml_to_destination_file(self):
+        """Safely saves formatted YML data to destination file."""
+        print(F'Saving output YML file to {self.output_file_name}')
+
+        # Configure safe dumper (multiline for strings)
+        yaml.SafeDumper.org_represent_str = yaml.SafeDumper.represent_str
+
+        def repr_str(dumper, data):
+            if '\n' in data:
+                return dumper.represent_scalar(u'tag:yaml.org,2002:str', data, style='|')
+            return dumper.org_represent_str(data)
+        yaml.add_representer(str, repr_str, Dumper=yamlordereddictloader.SafeDumper)
+
+        with open(self.output_file_name, 'w') as f:
+            yaml.dump(
+                self.yml_data,
+                f,
+                Dumper=yamlordereddictloader.SafeDumper,
+                default_flow_style=False)
+
+    def update_yml(self):
+        """Manager function for the generic YML updates."""
+        print_color(F'=======Starting updates for YML: {self.source_file}=======', LOG_COLORS.YELLOW)
+
+        self.remove_copy_and_dev_suffixes_from_name()
+        self.update_id_to_equal_name()
+        self.set_version_to_default()
+
+        print_color(F'=======Finished generic updates for YML: {self.output_file_name}=======', LOG_COLORS.YELLOW)
+
+    def initiate_file_validator(self, validator_type):
+        print_color('Starting validating files structure', LOG_COLORS.GREEN)
+
+        structure = StructureValidator(str(self.output_file_name))
```

Comment:
this will not work if the user uses some self created path, we should determine this by the type of the format happening at the moment

Reply:
and why did you remove the other validation? the script/PB/integration ones